library/formatting.inc.php

```diff
@@ -182,7 +182,7 @@ function TimeToHHMMSS($TimeValue)
         return '';
     }
 
-    $is_pm = (stripos($TimeValue, 'PM') !== false);
+    $is_pm = (stripos($TimeValue, 'pm') !== false);
```

Comment:
actually this shouldn't be needed since is a case insensitive search

what's weird in my testing is that `$TimeValue` below is only saved properly when stepping though the `if ($is_pm)` block, if not in debug mode then it's unable to compute the new `DateTime()`